hs_file_types/models/fileset.py

```diff
@@ -0,0 +1,186 @@
+import os
+import logging
+
+from django.db import models
+
+from hs_core.models import ResourceFile
+from base import AbstractLogicalFile
+from generic import GenericFileMetaDataMixin
+
+
+class FileSetMetaData(GenericFileMetaDataMixin):
+    pass
+
+
+class FileSetLogicalFile(AbstractLogicalFile):
+    """ One more files in a specific folder can be part of this aggregation """
+
+    metadata = models.OneToOneField(FileSetMetaData, related_name="logical_file")
+    # folder path relative to {resource_id}/data/contents/ that represents this aggregation
+    # folder becomes the name of the aggregation
+    folder = models.CharField(max_length=4096)
+    data_type = "GenericData"
+
+    @classmethod
+    def create(cls, resource):
+        # this custom method MUST be used to create an instance of this class
+        generic_metadata = FileSetMetaData.objects.create(keywords=[])
+        # Note we are not creating the logical file record in DB at this point
+        # the caller must save this to DB
+        return cls(metadata=generic_metadata, resource=resource)
+
+    @staticmethod
+    def get_aggregation_display_name():
+        return 'File Set Content: One or more files with specific metadata'
+
+    @staticmethod
+    def get_aggregation_type_name():
+        return "FileSetAggregation"
+
+    # used in discovery faceting to aggregate native and composite content types
+    @staticmethod
+    def get_discovery_content_type():
+        """Return a human-readable content type for discovery.
+        This must agree between Composite Types and native types (there is no equivalent native type
+        for File Set).
+        """
+        return "File Set"
+
+    @property
+    def can_contain_folders(self):
+        """This aggregation can contain folders"""
+        return True
+
+    @property
+    def supports_nested_aggregations(self):
```

Comment:
This property isn't being used anywhere

Reply:
and I don't think it exists on AbstractLogicalFile

Reply:
It exits in AbstractLogicalFile. However, I will remove this property as it is not used anywhere at this time.